### PR TITLE
Polling

### PR DESCRIPTION
**Pull request by titu**
BONUS: polling on each minute to keep syncing the courses from Canva to Differ

NOTE: this PR contains the changes made in the past PRs for ['courses' endpoint](https://github.com/titu/canvas-consumer/pull/8), ['import' endpoint](https://github.com/titu/canvas-consumer/pull/7), ['link-account' endpoint](https://github.com/titu/canvas-consumer/pull/6), [Routing](https://github.com/titu/canvas-consumer/pull/5) and [DB Connect](https://github.com/titu/canvas-consumer/pull/4)

### CODE REVIEW

File: config/settings.js [hunk 1]
```diff
@@ -0,0 +1,3 @@
+module.exports = {
+    syncTime: 60000 //one Minute
```
Comment (arealmaas) [inline config/settings.js]: separate 60000 to a constant:
const DEFAULT_SYNC_INTERVAL_IN_MILLISECONDS = 60000;

and change to use:
```
module.exports = {
   syncTime: process.env.SYNC_INTERVAL_IN_MILLISECONDS || DEFAULT_SYNC_INTERVAL_IN_MILLISECONDS
}
```

That way we can change the value through environment variables

File: api/sync.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+const settingsConfig = require('../config').settings;
```
Comment (arealmaas) [inline api/sync.js:1]: will set a new interval on every call to the API, so no need to expose this through an API, just start the interval when booting up the app for the first time.

**Review by arealmaas — CHANGES_REQUESTED** [review]

File: config/settings.js (hunk 1 above)
Comment (titu) [reply config/settings.js]: 👍

File: api/sync.js (hunk 2 above)
Comment (titu) [reply api/sync.js:1]: 👍

**Pull request merged**